### PR TITLE
feat(misc): automation code testing with docs file

### PR DESCRIPTION
**Pull request by gautamgambhir97**
-automation code testing of docs file

### CODE REVIEW

File: tests/automation_code.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            matches = re.finditer(pattern, markdown_content)
+            for match in matches:
+                code_content = match.group(2)
+            with open(temp_file_path, "w") as temp_file:
+                temp_file.write(code_content)
+            result = subprocess.Popen(
+                ["python", temp_file_path, "--port", "8000"])
+            print("Python script output:")
+            print(result.stdout)
+            time.sleep(10)
```
Comment (devjsc) [inline tests/automation_code.py]: why sleep?

File: tests/automation_code.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    print(f"Terminating process {proc.pid} using port {port}")
+                    process = psutil.Process(proc.pid)
+                    process.terminate()
+                    process.wait()
+        except (psutil.NoSuchProcess, psutil.AccessDenied, psutil.ZombieProcess):
+            pass
+
+
+class TestFetchAgents:
+    def test_creating_yeour_first_uAgent(self):
```
Comment (devjsc) [inline tests/automation_code.py]: spelling err

File: tests/automation_code.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import subprocess
+import os
+import re
+import tempfile
+import os
+import socket
+import psutil
+
+
+with open("pages/guides/agents/create-a-uagent.mdx", "r") as file:
```
Comment (devjsc) [inline tests/automation_code.py]: I think this needs to be more than this; 

```

import os
rootdir = 'docs/'

for subdir, dirs, files in os.walk(rootdir):
    for file in files:
        // with open file
        // look for code blocks in..
```

So we write something generic, and not per a file?

Comment (gautamgambhir97) [reply tests/automation_code.py]: Yes, we will do that, as we have just given you one example approach where we are extracting the code from docs. I will create one generic function for this.

File: tests/automation_code.py (hunk 1 above)
Comment (gautamgambhir97) [reply tests/automation_code.py]: we have use this because agent take some time to register on almanac

Comment (devjsc) [reply tests/automation_code.py]: Thanks for explaining, makes sense. But, this could take a long time to test everything using registration 😅

**Pull request closed without merging**